### PR TITLE
[SPARK-19591][ML][PYSPARK][FOLLOWUP] Add sample weights to decision trees

### PR DESCRIPTION
**Pull request by zhengruifeng**
## What changes were proposed in this pull request?
Add sample weights to decision trees

## How was this patch tested?
updated testsuites

### CODE REVIEW

**Comment by zhengruifeng** [comment]
retest this please

**Review by imatiach-msft — APPROVED** [review]
LGTM!

**Comment by imatiach-msft** [comment]
one concern that I'm not sure about, why does the build state:
```
* This patch adds the following public classes _(experimental)_:
* `class RandomForestClassifier(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredictionCol, HasSeed,`
* `class RandomForestRegressor(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredictionCol, HasSeed`
```
I'm not sure why it's showing this as they were not modified.

**Comment by imatiach-msft** [comment]
GBTClassifier and GBTRegressor don't support weight column yet.  There is a separate JIRA for random forest:
https://issues.apache.org/jira/browse/SPARK-9478
and another one for GBTs:
https://issues.apache.org/jira/browse/SPARK-9612
In GBTClassifier/Reg we currently don't pass weights:
https://github.com/apache/spark/blob/master/mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala#L204
Unfortunately GBTClassifier/Reg uses the same params as DecisionTreeClassifier/Reg, so this is kind of a bug now - since we expose this param but it doesn't actually do anything.  Thanks for finding this!
We should try to fix this soon at the scala layer but maybe in python for now we shouldn't expose it?

**Comment by zhengruifeng** [comment]
retest this please

File: python/pyspark/ml/classification.py [hunk 1]
```diff
@@ -902,12 +902,12 @@ class DecisionTreeClassifier(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPred
     >>> from pyspark.ml.linalg import Vectors
     >>> from pyspark.ml.feature import StringIndexer
     >>> df = spark.createDataFrame([
-    ...     (1.0, Vectors.dense(1.0)),
-    ...     (0.0, Vectors.sparse(1, [], []))], ["label", "features"])
+    ...     (1.0, 1.0, Vectors.dense(1.0)),
```
Comment (srowen) [inline python/pyspark/ml/classification.py]: My only nit here is that now this shows usage with weights, only, but the weight is 1. Maybe have two quick examples, one without specifying weight and one with a weight != 1?

**Comment by imatiach-msft** [comment]
strange, why does the build still say:
"* This patch adds the following public classes _(experimental)_:
* `class GBTClassifier(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredictionCol,`"
could this be a bug in the build system for python when there are multiple commits in a PR?

**Comment by zhengruifeng** [comment]
@srowen The test suites were updated by using non-one weights.
@imatiach-msft Now the build info is OK.

File: python/pyspark/ml/regression.py [hunk 2]
```diff
@@ -791,19 +792,34 @@ class DecisionTreeRegressor(JavaEstimator, HasFeaturesCol, HasLabelCol, HasPredi
     >>> model.transform(test1).head().variance
     0.0
 
+    >>> df3 = spark.createDataFrame([
```
Comment (srowen) [inline python/pyspark/ml/regression.py]: I guess I mean should we maintain an example without weights too? that might be the more common case. They don't both have to print depth, numNodes, etc.

Comment (zhengruifeng) [reply python/pyspark/ml/regression.py]: There already exists an example without weights. In the PR, I added examples with weights.
I think it is fine to keep those both. And I am OK to remove some info printed in the added ones.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**